### DISCUSSION TITLE
Errors in structured form?

### DISCUSSION THREAD

**Discussion by mmontone** [peggyjs/peggy#541] [Q&A] [answered]
Hi. Is it possible to get errors in a structured form? Similar to here: https://stackoverflow.com/a/26919119/4249433

```javascript
{
  "message": "Expected \"AND\", \"ORDER BY\" or end of input but \"A\" found.",
  "expected": [
    {
      "type": "literal",
      "value": "AND",
      "description": "\"AND\""
    },
    {
      "type": "literal",
      "value": "ORDER BY",
      "description": "\"ORDER BY\""
    },
    {
      "type": "end",
      "description": "end of input"
    }
  ],
  "found": "A",
  "offset": 9,
  "line": 1,
  "column": 10,
  "name": "SyntaxError"
}
```

I'm considering using Peggy for auto-completion and would need this to implement.

Thanks!

**Comment by Mingun** [comment, marked as answer]
Emm?? Errors always was in structured form:
https://github.com/peggyjs/peggy/blob/4036dabd40bdb318876ba0f3ac6fd5b0ae317176/lib/peg.d.ts#L639-L675

**Comment by mmontone** [comment]
Perfect. Thanks. Should have tried it. Apologies.
